movie/src/components/ErrorBoundary/ErrorBoundary.jsx

```diff
@@ -0,0 +1,18 @@
+import { Component } from "react";
+
+export default class ErrorBoundary extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { hasError: false };
+  }
+  componentDidCatch() {
+    this.setState({hasError: true})
+  }
```

Comment:
Look at the docs and what lifecycle method is using to update the state for next render.

> "componentDidCatch() is called during the “commit” phase, so side-effects are permitted. It should be used for things like logging errors:

https://reactjs.org/docs/error-boundaries.html#introducing-error-boundaries
https://reactjs.org/docs/react-component.html#componentdidcatch

Reply:
getDerivedStateFromError?

Reply:
Yes, but with static field.